src/summary_statistics/means.rs

```diff
@@ -250,6 +307,48 @@ mod tests {
             expected_geometric_mean,
             epsilon = 1e-12
         );
+
+        // Same as mean
+        let weights = Array1::from_elem(a.len(), 1.0 / a.len() as f64);
```

Comment:
It would be really cool to have both this check and the one below as standalone tests, ideally using `quickcheck`.

Reply:
I tried fixing this, but it's less "sexy" than I thought. Mostly because of the minimum size. I do agree that the tests are better now though.